wsi/drm/drmdisplay.cpp

```diff
@@ -409,6 +409,59 @@ bool DrmDisplay::GetDisplayAttribute(uint32_t config /*config*/,
   return status;
 }
 
+uint32_t DrmDisplay::GetPreferedDisplayMode(size_t modes_size) {
+  if (modes_size > 1) {
+    SPIN_LOCK(display_lock_);
+    for (size_t i = 0; i < modes_size; i++) {
+      // There is only one preferred mode per connector.
+      if (modes_[i].type & DRM_MODE_TYPE_PREFERRED) {
+        prefer_display_mode_ = i;
```

Comment:
Suggest don't update internal state in GetPreferedDisplayMode since the method name is more likely a read only method.